Templates/CDS-CSharp/SecurityHelper.csx

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.Remoting.Messaging;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.CommonDataService;
+using Microsoft.CommonDataService.Configuration;
+using Microsoft.CommonDataService.ServiceClient.Security;
+using Microsoft.CommonDataService.ServiceClient.Security.Credentials;
+
+internal class SecurityHelper
+{
+    public static Task<Client> GetClientFromConfiguration(HttpRequestMessage requestMessage, TraceWriter logger)
+    {
+        var settings = new ConnectionSettings
+        {
+            Tenant = EnvironmentValues.Tenant,
+            EnvironmentId = EnvironmentValues.EnvironmentId,
+            Credentials = new UserImpersonationCredentialsSettings()
+            {
+                ApplicationId = EnvironmentValues.ApplicationId,
+                RedirectUri = EnvironmentValues.RedirectUri,
+                ApplicationSecret = EnvironmentValues.ApplicationSecret
+            }
+        };
+
+        return settings.CreateClient(requestMessage, logger as TraceWriterTelemetryBridge ?? new TraceWriterTelemetryBridge());
```

Comment:
What does "ambient context" mean in this case? The EnvVar look like appsetting values, not per-request values. 

---
In reply to: [110759417](https://github.com/Azure/azure-webjobs-sdk-templates/pull/423#discussion_r110759417) [](ancestors = 110759417)